backend/src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -302,13 +331,54 @@
                 if (props != null) {
                     Endpoint endpoint = endpointsMap.get(props.getId());
                     endpoint.setProperties(props);
+                    // Todo: NOTIF-429 backward compatibility change - Remove soon.
+                    if (typedEndpointClass.equals(CamelProperties.class)) {
+                        endpoint.getProperties(CamelProperties.class).setSubType(endpoint.getSubType());
+                    }
                 }
             })).replaceWith(Uni.createFrom().voidItem());
         }
 
         return Uni.createFrom().voidItem();
     }
 
+    private <T extends EndpointProperties> Uni<List<T>> find(Class<T> typedEndpointClass, Set<UUID> endpointIds, boolean useStatelessSession) {
+        if (useStatelessSession) {
+            String query = "FROM " + typedEndpointClass.getSimpleName() + " WHERE id IN (:endpointIds)";
+            return sessionFactory.withStatelessSession(statelessSession -> {
+                return statelessSession.createQuery(query, typedEndpointClass)
+                        .setParameter("endpointIds", endpointIds)
+                        .getResultList();
+            });
+        } else {
+            return sessionFactory.withSession(session -> {
+                return session.find(typedEndpointClass, endpointIds.toArray());
+            });
+        }
+    }
+
+    private QueryBuilder<Endpoint> queryBuilderEndpointsPerType(String accountId, Set<CompositeEndpointType> type, Boolean activeOnly) {
+        Set<EndpointType> basicTypes = type.stream().filter(c -> c.getSubType() == null).map(CompositeEndpointType::getType).collect(Collectors.toSet());
+        Set<CompositeEndpointType> compositeTypes = type.stream().filter(c -> c.getSubType() != null).collect(Collectors.toSet());
+        return QueryBuilder
+                .builder(Endpoint.class)
+                .alias("e")
+                .where(
+                        WhereBuilder.builder()
+                            .ifElse(
+                                    accountId == null,
+                                    WhereBuilder.builder().and("e.accountId IS NULL"),
+                                    WhereBuilder.builder().and("e.accountId = :accountId", "accountId", accountId)
+                            )
+                            .and(
+                                    WhereBuilder.builder()
+                                            .ifOr(basicTypes.size() > 0, "e.compositeType.type IN (:endpointType)", "endpointType", basicTypes)
+                                            .ifOr(compositeTypes.size() > 0, "e.compositeType IN (:compositeTypes)", "compositeTypes", compositeTypes)
```

Comment:
I'm curious, how does that line translate into SQL? Do you have an example?

Reply:
I'll add it as a test to see the actual SQL

Reply:
You can see the results here: https://github.com/RedHatInsights/notifications-backend/pull/923/files#diff-a75040d7ad093bffe7edab6fc316b111646bdbb1f132ea0f2dc7bddbb5b74fccR22

Reply:
Thanks but that's the HQL, I was curious about the SQL 😃

Reply:
Ah! I can get it from psql too

Reply:
Perfect, thank you!